src/m3ninx/index/regexp.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package index
+
+import (
+	"bytes"
+	re "regexp"
+
+	vregex "github.com/couchbase/vellum/regexp"
+)
+
+// CompileRegex compiles the provided regexp.
+func CompileRegex(r []byte) (CompiledRegex, error) {
```

Comment:
Hey as we briefly discussed, could we instead use validation and just ensure that callers have fully anchored their regexps?

It is more explicit that way, rather than coercing their regexps into being fully anchored.  

I don't mind too much either way, but that seems less magical just responding with an error saying "hey please fully anchor your regexp with "^pattern$".

Reply:
done in https://github.com/m3db/m3/pull/902